### PR TITLE
target as list with radio-buttons

### LINKED ISSUE #346: Redesign setting target ROI
I suggest to replace drop-down list with checkboxes.
When plan is created there should be no checkbox selected by default, so user will be forced to choose something, instead of relying on preselected value.

### PR DESCRIPTION
**Pull request by ljelen**
closes #346

### CODE REVIEW

**Comment by grzanka** [comment]
User is still permitted to make no selection of Target ROI and Click OK. Afterwards user can create a field, click execute and get ugly error:
```
(pytripgui_dicom_target) grzanka@grzanka-VirtualBox:~$ pytripgui 
/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/viewcanvas_vc/viewcanvas_view.py:306: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  cax = self.axes.figure.add_axes([0.1, 0.1, 0.03, 0.8])
WARNING:pytripgui.plan_vc.plan_view:Given kernel wasn't found on the list
6
Traceback (most recent call last):
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/main_window_qt_vc/main_window_cont.py", line 68, in on_execute_selected_plan
    sim_results = self.app_callback.execute_plan(
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/app_logic/app_callbacks.py", line 84, in execute_plan
    item.data = self.executor.execute(patient.data, plan.data)
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/plan_executor/executor.py", line 35, in execute
    te.execute(plan)
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/execute.py", line 139, in execute
    plan.save_exec(plan._exec_path)  # add the .exec as well
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/plan.py", line 263, in save_exec
    self.make_exec()
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/plan.py", line 336, in make_exec
    output.extend(self._make_exec_input())  # input data, CT and VDX
  File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/plan.py", line 414, in _make_exec_input
    _name = self.voi_target.name.replace(" ", "_")
AttributeError: 'NoneType' object has no attribute 'name'
Aborted (core dumped)
```

**Comment by grzanka** [comment]
@ljelen can you also vertically align list of target ROIs to list of OARs ?
image

File: pytripgui/plan_vc/plan_view.py [hunk 1]
```diff
@@ -61,17 +61,25 @@ def uuid(self):
...
-        self.ui.targetROI_comboBox.setCurrentIndex(index_of_target)
+        item = QListWidgetItem(self.ui.targetROI_listWidget)
+        item.setData(Qt.UserRole, target)
+
+        self.ui.targetROI_listWidget.setItemWidget(item, target_item)
 
     def get_selected_target_roi(self):
-        return self.ui.targetROI_comboBox.currentData()
+        item_count = self.ui.targetROI_listWidget.count()
+        print(item_count)
```
Comment (grzanka) [inline pytripgui/plan_vc/plan_view.py:77]: remove debugging printout

File: pytripgui/plan_vc/plan_view.py [hunk 2]
```diff
@@ -61,17 +61,25 @@ def uuid(self):
...
+        for i in range(item_count):
```
Comment (grzanka) [inline pytripgui/plan_vc/plan_view.py:78]: Are you force to use iteration with index `i` ? I suspect there should be some iterator available in `self.ui.targetROI_listWidget`

**Review by grzanka — CHANGES_REQUESTED** [review]
Fix it

**Comment by ljelen** [comment]
> User is still permitted to make no selection of Target ROI and Click OK. Afterwards user can create a field, click execute and get ugly error:
> 
> ```
> (pytripgui_dicom_target) grzanka@grzanka-VirtualBox:~$ pytripgui 
> /home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/viewcanvas_vc/viewcanvas_view.py:306: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
>   cax = self.axes.figure.add_axes([0.1, 0.1, 0.03, 0.8])
> WARNING:pytripgui.plan_vc.plan_view:Given kernel wasn't found on the list
> 6
> Traceback (most recent call last):
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/main_window_qt_vc/main_window_cont.py", line 68, in on_execute_selected_plan
>     sim_results = self.app_callback.execute_plan(
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/app_logic/app_callbacks.py", line 84, in execute_plan
>     item.data = self.executor.execute(patient.data, plan.data)
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytripgui/plan_executor/executor.py", line 35, in execute
>     te.execute(plan)
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/execute.py", line 139, in execute
>     plan.save_exec(plan._exec_path)  # add the .exec as well
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/plan.py", line 263, in save_exec
>     self.make_exec()
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/plan.py", line 336, in make_exec
>     output.extend(self._make_exec_input())  # input data, CT and VDX
>   File "/home/grzanka/.conda/envs/pytripgui_dicom_target/lib/python3.8/site-packages/pytrip/tripexecuter/plan.py", line 414, in _make_exec_input
>     _name = self.voi_target.name.replace(" ", "_")
> AttributeError: 'NoneType' object has no attribute 'name'
> Aborted (core dumped)
> ```

image

File: pytripgui/messages.py [hunk 3]
```diff
@@ -32,7 +32,10 @@
         "Before continue, you should configure Trip98 paths in Settings/Trip98 Config"],
     "kernelSisPath": [
         "Given kernel has no sis path",
-        "Kernel selected by plan has no valid sis path"]
+        "Kernel selected by plan has no valid sis path"],
+    "noTargetRoiSelected": [
+        "No target ROI selected",
+        "Please select at least one TargetROI in 'Target' tab"]
```
Comment (grzanka) [inline pytripgui/messages.py:38]: This text doesn't make sense, as by design user can specify exactly one TargetROI. Why then adding `at least one` ?

**Review by grzanka — CHANGES_REQUESTED** [review]

**Review by grzanka — APPROVED** [review]

**Pull request merged**